internal/pkg/service/common/etcdop/watch.go

```diff
@@ -110,23 +117,21 @@ func (v Prefix) GetAllAndWatch(ctx context.Context, client *etcd.Client, handleE
 		})
 		sendBatch()
 
-		// Check iterator error
+		// Check getAll error
 		if err != nil {
-			initDoneCh <- err
-			close(initDoneCh)
+			outCh <- Events{InitErr: err}
 			return
 		}
 
 		// Continue with Watch where GetAll ended
-		close(initDoneCh)
```

Comment:
Problem bol tu, do `outCh` sme zapisali poslednu hodnotu a `initDoneCh` zavreli.

Lenze na druhej strane `outCh` je nejaky for, ktory s tou hodnotou nieco robi.
A to ze druha strana `outCh` tie `events` prijala, neznamena, ze ich aj spracoval.

Robilo to problem v `distribution` pkg, kde sa tento mechanizmus pouziva na discovery ostatnych worker nodes.

Pri starte potrebujeme pockat kym sa nacitaju vsetky existujuce `nodes` a az tak pokracovat.
Zmeny sa potom spracovavaju na pozadi.

Lenze my potrebujeme cakat, kym sa aktualizuje zoznam nodes v `distribution.Node`, nie iba na prijatie poslednej hodnoty z etcd. 

Teda `initDone` sa odblokoval, ale zoznam `nodes` este nebol kompletny, ... a potom to generovalo zmeny, ktore sa nestali, ... iba sa proste dodatocne spracovala posledna prijata hodnota z `GetAll`.

Reply:
Teda `initDone` v tejto podobe isiel prec.